README.md

```diff
@@ -1,19 +1,112 @@
+```
+$ cd ~/path/to/app
+```
+
+The first thing you'll want to do is create a `package.json` file which contains meta data about your app including a list of your apps dependencies. Remember: "dependencies" are just the modules that your app needs in order to function - In other words, dependencies are the modules or packages your app *depends* on.
+
+We can create a `package.json` file with ease by executing the following command in our project's root directory:
+```
+$ npm init
+```
+
+Executing this command will prompt you for a number of things, like the name and version of your application. For now, you can simply hit `return` to accept the defaults for most of the questions, with the following exception:
+```
+entry point: (index.js)
+```
+Just go ahead and enter the name of your main file then press `enter`
+
+Next we'll install the Express package:
+```
+$ npm install express --save
+```
+NPM will search the registry for the Express package and download it to the `node_modules` folder in your projects root directory. The `--save` part just tracks Express as a new dependency in your `package.json` file.
+
+Now we're ready to start building our simple Express app. Go ahead and copy this code into your `index.js`.
+
+`index.js:`
+```
+const express = require('express');
+const app = express();
+
+app.get('/', function (req, res) {
+  res.send('Hello World!');
+});
+
+app.listen(3000, function () {
+  console.log('Example app listening on port 3000!');
+});
+```
+
+The first thing you'll notice here is the use of `require()` to import  Express into our app.
+
+The `require()` function is part of Nodes built in module loading system which we can use to import JavaScript from external files into our application.
+
+Now that express is available to our main application, we can use its methods to helps us build our app. We're using the `listen()` method to listen for connections on port 3000 as well as some routing functionality to send the message `Hello World` to the client when the index `/` route is requested.
+
+Using `require()` to import Express from another location within our app allows us to expand the capability of our main application logic in a way that's clean and manageable.
+
+## Creating your own modules
+NPM is full of packages you can use in your app to speed up development and build great things.
+
+But what if you wanted to build your own modules? Great question!
+
+Let's say you need to write some JavaScript code that performs calculations on shapes. Rather than writing this code in your main application, it would be a good idea to bundle all this related code into it's own module file that we can import just like we did with Express.
+
+Let's create a module file called `circle.js`, paste in the following code and save it to our projects root directory:
+
+`circle.js:`
+```
+const PI = Math.PI;
+
+exports.area = (r) => PI * r * r;
+
+exports.circumference = (r) => 2 * PI * r;
+
+```
```

Comment:
This is great! I wonder if moving this section up might help students understand the Express bits a bit more easily?